src/general/Hex.cs

```diff
@@ -231,19 +231,50 @@ public static Hex FlipHorizontally(Hex hex)
         return new Hex(-hex.Q, hex.Q + hex.R);
     }
 
-    public override bool Equals(object obj)
+    /// <summary>
+    ///   Returns the RenderPriority for the hex.
+    ///   Currently between 1 and 42.
+    ///   This assures 4 distance before the same RenderIndex.
+    /// </summary>
+    /// <returns>RenderPriority</returns>
+    public static int GetRenderPriority(Hex hex)
     {
-        if (!(obj is Hex hex))
-            return false;
+        int radius = (Math.Abs(hex.Q) + Math.Abs(hex.Q + hex.R) + Math.Abs(hex.R)) / 2;
+        if (radius == 0)
+            return 1;
 
-        return Equals(hex);
+        var currentHex = HexNeighbourOffset[HexSide.BottomLeft] * radius;
+        int indexInRing = 0;
+        for (int i = 0; i < 6; ++i)
+        {
+            for (int j = 0; j < radius; ++j)
+            {
+                if (currentHex == hex)
+                {
+                    return (radius % 6 + 1) * 6 + (indexInRing % 6 + 1);
+                }
+
+                indexInRing++;
+                currentHex += HexNeighbourOffset[(HexSide)(i + 1)];
+            }
+        }
```

Comment:
Can't this looping be fixed somehow? This seems to just look for a hex that matches the current given position?
Why can't a direct computation from `hex` to render index without looping be done?